### ISSUE TITLE
"rs" does not work on osx

### ISSUE THREAD

**Issue by jeremieca** [remy/nodemon#511]
Hello,

I'm using Yosemite and node 0.12.0. When I enter "rs" command, nothing happened. Whereas with node 0.10.x I had no problem.

Thanks !

**Comment by remy** [comment]
I don't have Yosemite, but I can upgrade to 0.12.x. I'll give it a try.

**Comment by remy** [comment]
Hmm, works for me. What state doesn't this work for you?

I'm running a normal (node) process and `rs` restarts as per:

```text
» PORT=8001 nodemon
26 Feb 13:44:39 - [nodemon] v1.3.7
26 Feb 13:44:39 - [nodemon] to restart at any time, enter `rs`
26 Feb 13:44:39 - [nodemon] watching: *.*
26 Feb 13:44:39 - [nodemon] starting `node lib/index.js`
rs
26 Feb 13:44:40 - [nodemon] starting `node lib/index.js`
clearing cache in development mode
autocache version: 0.4.5
setting domain .picturehouses.local
picturehouse v1.0.1 is up and running on port http://localhost:8001
rs
26 Feb 13:44:41 - [nodemon] starting `node lib/index.js`
clearing cache in development mode
autocache version: 0.4.5
setting domain .picturehouses.local
picturehouse v1.0.1 is up and running on port http://localhost:8001
```

**Comment by jeremieca** [comment]
Ok... Thanks.

So, I don't know how to help you debug this problem ?

...

**Comment by remy** [comment]
You can start by pasting what you see, the same way I have.

**Comment by jeremieca** [comment]
So I test on a new project and Nodemon works well. In fact maybe the problem is not the node upgrade.

So my new question is : In a project what is able to block "rs" command ?
Restart on file save is ok.

I put my ouput bellow. On "rs", just nothing happen.

     > ./node_modules/nodemon/bin/nodemon.js server.js
     
     27 Feb 11:33:45 - [nodemon] v1.3.7
     27 Feb 11:33:45 - [nodemon] to restart at any time, enter `rs`
     27 Feb 11:33:45 - [nodemon] watching: *.*
     27 Feb 11:33:45 - [nodemon] starting `node server.js`
     Failed to load c++ bson extension, using pure JS version
     info: Static routes launched
     info: Server launched on port 8080
     info: Websocket listening
     App ready - Mode dev
     
     rs

**Comment by remy** [comment]
Can you also post the output from `./node_modules/nodemon/bin/nodemon.js server.js --dump`?

**Comment by jeremieca** [comment]
27 Feb 12:39:55 - [nodemon] v1.3.7
27 Feb 12:39:55 - [nodemon] to restart at any time, enter `rs`
27 Feb 12:39:55 - [nodemon] watching: *.*
--------------
node: v0.12.0
nodemon: v1.3.7
command: node /Volumes/DataHD/R&D/Personnel/Deescut/Developpement/Core.node/project/node_modules/nodemon/bin/nodemon.js server.js --dump
cwd: /Volumes/DataHD/R&D/Personnel/Deescut/Developpement/Core.node/project
OS: darwin x64
--------------
{ run: false,
  system: 
   { cwd: '/Volumes/DataHD/R&D/Personnel/Deescut/Developpement/Core.node/project',
     useFind: true,
     useWatch: false,
     useWatchFile: false },
  required: false,
  dirs: [ '/Volumes/DataHD/R&D/Personnel/Deescut/Developpement/Core.node/project' ],
  timeout: 1000,
  options: 
   { dump: true,
     ignore: 
      [ '.git',
        'node_modules',
        'bower_components',
        '.sass-cache',
        re: /\.git|node_modules|bower_components|\.sass\-cache/ ],
     watch: [ '*.*', re: /.*\..*/ ],
     restartable: 'rs',
     execMap: { py: 'python', rb: 'ruby' },
     stdin: true,
     verbose: false,
     stdout: true,
     execOptions: 
      { script: 'server.js',
        exec: 'node',
        args: [],
        scriptPosition: 0,
        nodeArgs: undefined,
        ext: 'js',
        env: {},
        execArgs: [] },
     monitor: 
      [ '*.*',
        '!.git',
        '!/Volumes/DataHD/R&D/Personnel/Deescut/Developpement/Core.node/project/node_modules/**/*',
        '!bower_components',
        '!.sass-cache' ] },
  load: [Function],
  reset: [Function: reset],
  lastStarted: 0,
  loaded: [],
  command: 
   { raw: { executable: 'node', args: [ 'server.js' ] },
     string: 'node server.js' },
  offset: 0 }
--------------

**Comment by Xiot** [comment]
I ran into the same thing as well on my Windows 8 x64 machine.

If the process crashes, typing `rs` appears to do nothing.
However if I type `rs` while everything is up and running, I get this
```
rs
exception in nodemon killing node
Error: write after end
    at writeAfterEnd (_stream_writable.js:167:12)
    at Socket.Writable.write (_stream_writable.js:214:5)
    at Socket.write (net.js:625:40)
    at ReadStream.ondata (_stream_readable.js:540:20)
    at ReadStream.emit (events.js:129:20)
    at readableAddChunk (_stream_readable.js:163:16)
    at ReadStream.Readable.push (_stream_readable.js:126:10)
    at TTY.onread (net.js:529:20)

----------------------------------------------------------
If appropriate, please file an error with the output from:
$ D:\Program Files\nodejs\\node.exe D:\Program Files\nodejs\node_modules\nodemon\bin\nodemon.js --debug --dump
At http://github.com/remy/nodemon/issues/new
----------------------------------------------------------
```

Here is my dump
```
--------------
node: v0.12.0
nodemon: v1.3.6
command: D:\Program Files\nodejs\node.exe D:\Program Files\nodejs\node_modules\nodemon\bin\nodemon.js --debug --dump
cwd: D:\Development\Git Hub\genie\magic-lamp
OS: win32 x64
--------------
{ run: false,
  system:
   { cwd: 'D:\\Development\\Git Hub\\genie\\magic-lamp',
     useFind: false,
     useWatch: false,
     useWatchFile: true },
  required: false,
  dirs: [ 'D:\\Development\\Git Hub\\genie\\magic-lamp' ],
  timeout: 1000,
  options:
   { dump: true,
     execMap1: { js: 'node' },
     restartable: 'rs',
     env:
      { DEBUG: 'magic-lamp*,mongoose-adapter',
        PORT: '3000',
        ETC: 'etc' },
     ignore:
      [ '.git',
        'node_modules',
        'bower_components',
        '.sass-cache',
        re: /\.git|node_modules|bower_components|\.sass\-cache/ ],
     watch: [ '*.*', re: /.*\..*/ ],
     execMap: { py: 'python', rb: 'ruby' },
     stdin: true,
     verbose: false,
     stdout: true,
     execOptions:
      { script: './server.js',
        exec: 'node',
        args: [ '--debug' ],
        scriptPosition: 1,
        nodeArgs: undefined,
        ext: 'js,es,es6',
        env:
         { DEBUG: 'magic-lamp*,mongoose-adapter',
           PORT: '3000',
           ETC: 'etc' },
        execArgs: [] },
     monitor:
      [ '*.*',
        '!.git',
        '!D:\\Development\\Git Hub\\genie\\magic-lamp\\node_modules/**/*',
        '!bower_components',
        '!.sass-cache' ] },
  load: [Function],
  reset: [Function: reset],
  lastStarted: 0,
  loaded: [ 'D:\\Development\\Git Hub\\genie\\magic-lamp\\nodemon.json' ],
  command:
   { raw: { executable: 'node', args: [ '--debug', './server.js' ] },
     string: 'node --debug ./server.js' } }
--------------
```

**Comment by jeremieca** [comment]
I try to find the bug and I discovered very strange behaviour. When I open in the same folder 10 terminal with my project, in some tab 'rs' always works. In others tab, 'rs' does not work... I really don't understand. So I fix my problem using a new tab of my terminal...

Thanks.

**Comment by cedx** [comment]
Same problem here: "rs" not working anymore after upgrade to OSX Yosemite. I also did an upgrade of Node.js to v0.12 but the issue raised before the upgrade (e.g. with v0.10). My watched process is a PHP script that exits cleanly.

I don't know if the problem is related to Nodemon or to OSX Yosemite (or the shipped PHP version). Opening a new terminal tab doesn't change anything.
 
My dump output:
```
5 Mar 13:51:55 - [nodemon] v1.3.7
5 Mar 13:51:55 - [nodemon] to restart at any time, enter `rs`
5 Mar 13:51:55 - [nodemon] watching: *.*
--------------
node: v0.12.0
nodemon: v1.3.7
command: node /usr/local/lib/node_modules/nodemon/bin/nodemon.js --dump
cwd: /Users/cedric
OS: darwin x64
--------------
{ run: false,
  system: 
   { cwd: '/Users/cedric',
     useFind: true,
     useWatch: false,
     useWatchFile: false },
  required: false,
  dirs: [ '/Users/cedric' ],
  timeout: 1000,
  options: 
   { dump: true,
     ignore: 
      [ '.git',
        'node_modules',
        'bower_components',
        '.sass-cache',
        re: /\.git|node_modules|bower_components|\.sass\-cache/ ],
     watch: [ '*.*', re: /.*\..*/ ],
     restartable: 'rs',
     execMap: { py: 'python', rb: 'ruby' },
     stdin: true,
     verbose: false,
     stdout: true,
     execOptions: 
      { script: null,
        exec: 'node',
        args: [],
        scriptPosition: null,
        nodeArgs: undefined,
        ext: 'js',
        env: {},
        execArgs: [] },
     monitor: 
      [ '*.*',
        '!.git',
        '!node_modules',
        '!bower_components',
        '!.sass-cache' ] },
  load: [Function],
  reset: [Function: reset],
  lastStarted: 0,
  loaded: [],
  command: { raw: { executable: 'node', args: [] }, string: 'node' },
  offset: 0 }
--------------
```

**Comment by SamDecrock** [comment]
I have the same problem on Mavericks. Using node.js 0.12.0

If I hit ENTER a second time, it works!

**Comment by JacksonGariety** [comment]
I'm having this same issue (where sometimes rs works and others not) and it was vexing to me for ~20 minutes. I eventually realized that under Node `0.12`, `rs` only works prior to a clean exit, but not after. ON Node 0.10.26 it works either way. Would you re-open this, @remy?

**Comment by danfooo** [comment]
`rs` seems to fail me about 1/4 times. I'm on node 0.12.0. I tried to debug a bit and saw that when it fails, the callback to `psTree` in `lib/monitor/run` isn't run. [That](https://github.com/indexzero/ps-tree/blob/master/index.js#L14) is sadly where it loses me though..

**Comment by remy** [comment]
I've not been able to replicate this myself (and I use nodemon in multiple daily), but @JacksonGariety's insights might be what I need to replicate.

**Comment by hansjd** [comment]
i'm having the same issue... for me it's only when nodemon is waiting for file change.

1 Apr 22:34:38 - [nodemon] starting `node server.js`
API is running on port 1337
rs
1 Apr 22:34:39 - [nodemon] starting `node server.js`
API is running on port 1337
rs
SELECT * FROM `customers` 
ReferenceError: result is not defined
    at Object.exports.select (/Users/hansjd/Dev/work-logger/mysqlConnection.js:29:14)
    at /Users/hansjd/Dev/work-logger/routes/customers.js:45:5
    at Layer.handle [as handle_request] (/Users/hansjd/Dev/work-logger/node_modules/express/lib/router/layer.js:82:5)
    at next (/Users/hansjd/Dev/work-logger/node_modules/express/lib/router/route.js:110:13)
    at Route.dispatch (/Users/hansjd/Dev/work-logger/node_modules/express/lib/router/route.js:91:3)
    at Layer.handle [as handle_request] (/Users/hansjd/Dev/work-logger/node_modules/express/lib/router/layer.js:82:5)
    at /Users/hansjd/Dev/work-logger/node_modules/express/lib/router/index.js:267:22
    at Function.proto.process_params (/Users/hansjd/Dev/work-logger/node_modules/express/lib/router/index.js:321:12)
    at next (/Users/hansjd/Dev/work-logger/node_modules/express/lib/router/index.js:261:10)
    at Function.proto.handle (/Users/hansjd/Dev/work-logger/node_modules/express/lib/router/index.js:166:3)
[ { customer_id: 1,
    customer_name: 'rebo',
    active: 1,
    created: '2015-04-01T13:30:53.000Z',
    updated: null,
    deleted: null } ]
/Users/hansjd/Dev/work-logger/node_modules/mysql/lib/protocol/Parser.js:82
        throw err;
              ^
Error: Can't set headers after they are sent.
    at ServerResponse.OutgoingMessage.setHeader (_http_outgoing.js:335:11)
    at ServerResponse.header (/Users/hansjd/Dev/work-logger/node_modules/express/lib/response.js:700:10)
    at ServerResponse.send (/Users/hansjd/Dev/work-logger/node_modules/express/lib/response.js:154:12)
    at ServerResponse.json (/Users/hansjd/Dev/work-logger/node_modules/express/lib/response.js:240:15)
    at ServerResponse.send (/Users/hansjd/Dev/work-logger/node_modules/express/lib/response.js:142:21)
    at /Users/hansjd/Dev/work-logger/routes/customers.js:47:7
    at Query._callback (/Users/hansjd/Dev/work-logger/mysqlConnection.js:26:3)
    at Query.Sequence.end (/Users/hansjd/Dev/work-logger/node_modules/mysql/lib/protocol/sequences/Sequence.js:96:24)
    at Query._handleFinalResultPacket (/Users/hansjd/Dev/work-logger/node_modules/mysql/lib/protocol/sequences/Query.js:143:8)
    at Query.EofPacket (/Users/hansjd/Dev/work-logger/node_modules/mysql/lib/protocol/sequences/Query.js:127:8)
rs
1 Apr 22:34:46 - [nodemon] app crashed - waiting for file changes before starting...
rs

**Comment by dogancelik** [comment]
I have the same issue in Windows 7 (64-bit)

When I type `rs` and press Enter, nothing happens. When I try to type `rs` again, I can't type anything. No character shows up on console.

Log:
```
D:\03. Projeler\Node\test>nodemon test.coffee --debug --dump
25 Apr 16:44:24 - [nodemon] v1.3.7
25 Apr 16:44:24 - [nodemon] to restart at any time, enter `rs`
25 Apr 16:44:24 - [nodemon] watching: *.*
--------------
node: v0.12.2
nodemon: v1.3.7
command: node C:\Users\dogan\AppData\Roaming\npm\node_modules\nodemon\bin\nodemon.js test.coffee --d
ebug --dump
cwd: D:\03. Projeler\Node\test
OS: win32 x64
--------------
{ run: false,
  system:
   { cwd: 'D:\\03. Projeler\\Node\\test',
     useFind: false,
     useWatch: false,
     useWatchFile: true },
  required: false,
  dirs: [ 'D:\\03. Projeler\\Node\\test' ],
  timeout: 1000,
  options:
   { dump: true,
     ignore:
      [ '.git',
        'node_modules',
        'bower_components',
        '.sass-cache',
        re: /\.git|node_modules|bower_components|\.sass\-cache/ ],
     watch: [ '*.*', re: /.*\..*/ ],
     restartable: 'rs',
     execMap: { py: 'python', rb: 'ruby' },
     stdin: true,
     verbose: false,
     stdout: true,
     execOptions:
      { script: 'test.coffee',
        exec: 'coffee.cmd',
        args: [ '--debug' ],
        scriptPosition: 0,
        nodeArgs: undefined,
        ext: 'coffee,litcoffee,js',
        env: {},
        execArgs: [ '--nodejs' ] },
     monitor:
      [ '*.*',
        '!.git',
        '!D:\\03. Projeler\\Node\\test\\node_modules/**/*',
        '!bower_components',
        '!.sass-cache' ] },
  load: [Function],
  reset: [Function: reset],
  lastStarted: 0,
  loaded: [],
  command:
   { raw:
      { executable: 'coffee.cmd',
        args: [ '--nodejs', 'test.coffee', '--debug' ] },
     string: 'coffee.cmd --nodejs test.coffee --debug' },
  offset: 0 }
--------------
```

**Comment by remy** [comment]
Can you debug it at all?

**Comment by enterframe** [comment]
same for me, OSX Yosemite, Vagrant -> Debian

20 Jun 23:43:27 - [nodemon] v1.3.7
20 Jun 23:43:27 - [nodemon] reading config /vagrant/node/nodemon.json
20 Jun 23:43:27 - [nodemon] to restart at any time, enter `rs`
20 Jun 23:43:27 - [nodemon] ignoring: node_modules/***/*
20 Jun 23:43:27 - [nodemon] watching: *.*
20 Jun 23:43:27 - [nodemon] watching extensions: js
--------------
node: v0.12.4
nodemon: v1.3.7
command: node /usr/lib/node_modules/nodemon/bin/nodemon.js --dump
cwd: /vagrant/node
OS: linux x64
--------------
{ run: false,
  system: 
   { cwd: '/vagrant/node',
     useFind: false,
     useWatch: true,
     useWatchFile: false },
  required: false,
  dirs: [ '/vagrant/node' ],
  timeout: 1000,
  options: 
   { dump: true,
     verbose: true,
     ignore: [ 'node_modules/**', re: /node_modules/.*.*/ ],
     watch: [ '*.*', re: /.*\..*/ ],
     restartable: 'rs',
     execMap: { py: 'python', rb: 'ruby' },
     stdin: true,
     stdout: true,
     execOptions: 
      { script: '',
        exec: 'grunt',
        args: [],
        scriptPosition: 0,
        nodeArgs: undefined,
        ext: 'js',
        env: {},
        execArgs: [] },
     monitor: [ '*.*', '!node_modules/***/*' ] },
  load: [Function],
  reset: [Function: reset],
  lastStarted: 0,
  loaded: [ '/vagrant/node/nodemon.json' ],
  command: { raw: { executable: 'grunt', args: [] }, string: 'grunt' },
  offset: 0 }
--------------
20 Jun 23:43:27 - [nodemon] exiting

**Comment by remy** [comment]
Again, debug please.

I'm able to do this - I do it all day long. Also, vagrant + debian is not OS X...

@enterframe it seems like you're using the grunt version too...please try without grunt.

**Comment by enterframe** [comment]
@remy sorry, it´s not osx, sure. "rs" without grunt-nodemon works

**Comment by remy** [comment]
Fixed in nodemon@1.6.1.
